tests/queries/0_stateless/01866_split_by_regexp.reference

```diff
@@ -9,12 +9,21 @@ Test fallback of splitByRegexp to splitByChar if regexp is trivial
 ['a','b','c']
 ['a','b','c']
 ['','','','','','']
-['a^b^c']
-['a$b$c']
-['a)b)c']
+['a','^','b','^','c']
+['a','$','b','$','c']
```

Comment:
I assume those two are incorrect, since `^` matches only beginning of string. 

Python produces:

```
re.split('^', 'abc')
Out[5]: ['', 'abc']
```

Reply:
Yes. Maybe keeping it as it was is a better choice. I can't find an simple way to fix it for now. Let's close this pr.